src/scripts/autocomplete.js

```diff
@@ -0,0 +1,509 @@
+        this.el.querySelector("li[title='" + ids + "']").classList.add("selected-" + this.type);
+      }
+    }
+  }
+};
+
+ProjectAutoComplete.prototype.selectTask = function (elem, silent) {
+  // Set selected task
+  var currentSelected = this.el.querySelector(".selected-task");
+
+  if (!!currentSelected) {
+    currentSelected.classList.remove("selected-task");
+  }
+  elem.classList.add("selected-task");
+
+  // Set selected project
+  this.selectProject(elem.parentNode.parentNode, silent, true);
+};
+
+ProjectAutoComplete.prototype.selectProject = function (elem, silent, removeTask) {
+  if (elem.classList.contains("item-name")) {
+    elem = elem.parentNode;
+  }
+  if (!elem.classList.contains(this.type + "-row")) {
+    if (elem.classList.contains("task-count")) {
+      this.toggleTaskList(elem);
+    }
+    if (elem.classList.contains("task-item")) {
+      this.selectTask(elem);
+    }
+    return;
+  }
+
+  if (!removeTask && this.el.querySelector(".selected-task")) {
+    this.el.querySelector(".selected-task").classList.remove("selected-task");
+  }
+
+  var currentSelected = this.el.querySelector(".selected-" + this.type),
+    val = elem.getAttribute("data-pid");
+
+  if (!!currentSelected) {
+    currentSelected.classList.remove("selected-" + this.type);
+  }
+  elem.classList.add("selected-" + this.type);
+
+  // Update placeholder
+  this.placeholderDiv.textContent = this.placeholderDiv.title = this.generateLabel(this.getSelected(), val, this.type);
+  this.setProjectBullet(val);
+
+  if (!silent) {
+    // Close dropdown
+    this.closeDropdown();
+  }
+  return false;
+};
+
+ProjectAutoComplete.prototype.getSelected = function () {
+  var selected = this.el.querySelector(".selected-" + this.type),
+    task = this.el.querySelector(".selected-task"),
+    pid = (!!selected) ? parseInt(selected.getAttribute("data-pid"), 10) : 0,
+    tid = (!!task) ? parseInt(task.getAttribute("data-tid"), 10) : null,
+    name = (!!selected) ? selected.getAttribute("title") : "";
+
+  return {
+    el: selected,
+    pid: pid,
+    tid: tid,
+    name: name
+  };
+};
+
+ProjectAutoComplete.prototype.getSelectedProjectByPid = function (pid) {
+  var selected = this.el.querySelector("li[data-pid='" + pid + "']"),
+    name = (!!selected) ? selected.textContent : "";
+
+  return {
+    el: selected,
+    pid: pid,
+    name: name
+  };
+};
+
+ProjectAutoComplete.prototype.setProjectBullet = function (pid, tid, el) {
+  var project,
+    className,
+    id = parseInt(pid, 10),
+    elem = el || this.placeholderItem.querySelector(".project-bullet"),
+    result,
+    task;
+
+  if (!!pid) {
+    project = this.el.querySelector("li[data-pid='" + pid + "']");
+    if (id !== 0) {
+      if (!!project) {
+        className = project.querySelector(".project-bullet").className;
+        elem.className = className;
+        result = " - " + project.getAttribute("title");
+        task = project.querySelector("li[data-tid='" + tid + "']");
```

Comment:
Some calling code checks if (!!pid), if (!!tid) and if not individually replaces them with 0 -- but some calls to `setProjectBullet` don't. Is it OK to use straight tid here?